apps/omg_eth/lib/omg_eth/release_tasks/set_contract.ex

```diff
@@ -85,20 +86,25 @@ defmodule OMG.Eth.ReleaseTasks.SetContract do
 
     txhash_contract = get_env(network <> "_TXHASH_CONTRACT")
     authority_address = get_env(network <> "_AUTHORITY_ADDRESS")
-    contract_address = get_env(network <> "_CONTRACT_ADDRESS")
+    env_contract_addresses = get_env(network <> "_CONTRACT_ADDRESS")
+    contract_addresses = if env_contract_addresses, do: Jason.decode!(env_contract_addresses), else: nil
 
     exit_period_seconds =
       validate_integer(get_env("EXIT_PERIOD_SECONDS"), Application.get_env(@app, :exit_period_seconds))
 
-    update_configuration(txhash_contract, authority_address, contract_address, exit_period_seconds)
+    update_configuration(txhash_contract, authority_address, contract_addresses, exit_period_seconds)
   end
 
-  defp update_configuration(txhash_contract, authority_address, contract_address, exit_period_seconds)
+  defp update_configuration(txhash_contract, authority_address, contract_addresses, exit_period_seconds)
        when is_binary(txhash_contract) and
-              is_binary(authority_address) and is_binary(contract_address) and is_integer(exit_period_seconds) do
+              is_binary(authority_address) and is_map(contract_addresses) and is_integer(exit_period_seconds) do
+    contract_addresses =
+      contract_addresses
+      |> Enum.into(%{}, fn {name, addr} -> {String.to_existing_atom(to_string(name)), String.downcase(addr)} end)
+
```

Comment:
@pdobacz are you sure `to_existing_atom` is ok? the VM has just booted, it's likely no atoms exist at all.

```suggestion
Enum.into(contract_addresses, %{}, fn {name, addr} -> {String.atom(to_string(name)), String.downcase(addr)} end)
```

also, there's no need to check is_binary, is_binary, is_map if you're not intending to provide a valid exit/1 clause. pls follow the convention by validating input parameters and exit/1 the process with a descriptive message.

Reply:
yeah, missing atoms seem to be the issue, and also an issue independent of how the system env variables are structured (packed in a JSON or separate).

Reply:
on the latter part:

> there's no need to check is_binary, is_binary, is_map if you're not intending to provide a valid exit

but the checks `is_binary` weren't introduced in this PR. I was just conforming to the checks I already found here (by switching to an `is_map` check). Or am I missing a point here?